src/core/lv_obj_tree.c

```diff
@@ -210,7 +210,9 @@ void lv_obj_move_to_index(lv_obj_t * obj, int32_t index)
     }
 
     const uint32_t parent_child_count = lv_obj_get_child_count(parent);
+    /* old_index only can be 0 or greater, this point can not be reached if the parent is null */
     const int32_t old_index = lv_obj_get_index(obj);
+    LV_ASSERT(0 <= old_index);
```

Comment:
Fail assertion if `old_index` is `-1`, this only happens  when parent is NULL, but this was checked above.